packages/dd-trace/src/plugins/database.js

```diff
@@ -1,9 +1,36 @@
 'use strict'
 
 const StoragePlugin = require('./storage')
+const TextMapPropagator = require('../opentracing/propagation/text_map.js')
+const Config = require('../../src/config')
+const config = new Config()
+const textMap = new TextMapPropagator(config)
+const carrier = {}
 
 class DatabasePlugin extends StoragePlugin {
   static get operation () { return 'query' }
+
+  createDBMPropagationCommentService () {
+    const dddbs = encodeURIComponent(this.config.service)
+    const dde = encodeURIComponent(this.tracer._env)
+    const ddps = encodeURIComponent(this.tracer._service)
+    const ddpv = encodeURIComponent(this.tracer._version)
+
+    return `dddbs='${dddbs}',dde='${dde}',ddps='${ddps}',ddpv='${ddpv}'`
+  }
+
+  injectDBMquery (query) {
+    config.experimental.traceparent = true
+    const serviceTags = this.createDBMPropagationCommentService()
+    textMap._injectTraceparent(this.activeSpan._spanContext, carrier)
+
+    if (this.config.dbmPropagationMode === 'service') {
+      return `/*${serviceTags}*/ ${query}`
+    } else if (this.config.dbmPropagationMode === 'full') {
+      return `/*${serviceTags},traceparent='${carrier.traceparent}'*/ ${query}`
```

Comment:
Note that `carrier.traceparent` is generated by us and is not at risk of having special characters which is why it bypasses `encodeURIComponent()`.